### ISSUE TITLE
AKS, GKE cluster provisioning errors out in UI

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#28179]
**What kind of request is this (question/bug/enhancement/feature request):** Bug

**Steps to reproduce (least amount of steps as possible):**
1. Create AKS or GKE cluster in UI
2. After hitting the "Create" button, the UI gets stuck in "Saving" state and after some time the below error is displayed
![Screen Shot 2020-07-30 at 12 47 57 PM](https://user-images.githubusercontent.com/18536626/88973936-b26cd800-d26c-11ea-839e-fb023c74210a.png)

But the provisioning actually starts and succeeds. The above error is seen in the UI during cluster creation. 
No specific errors observed in the UI network console.

![Screen Shot 2020-07-30 at 2 13 55 PM](https://user-images.githubusercontent.com/18536626/88975388-f19c2880-d26e-11ea-835c-5fa8e04bf260.png)

**Other details that may be helpful:**
The automation tests consistently fail for these clusters during provisioning. It fails during the check for cluster state validation where reload of cluster object fails. 

```
tests/v3_api/test_aks_cluster.py:33: in create_and_validate_aks_cluster
    skipIngresscheck=True)
tests/v3_api/common.py:897: in validate_cluster
    nodes_not_in_active_state=nodes_not_in_active_state)
tests/v3_api/common.py:1364: in validate_cluster_state
    timeout=MACHINE_TIMEOUT)
tests/v3_api/common.py:260: in wait_for_condition
    resource = client.reload(resource)
/usr/local/lib/python3.7/site-packages/rancher.py:430: in reload
    return self.by_id(obj.type, obj.id)
/usr/local/lib/python3.7/site-packages/rancher.py:363: in by_id
    raise e
/usr/local/lib/python3.7/site-packages/rancher.py:358: in by_id
    return self._get(url, self._to_dict(**kw))
/usr/local/lib/python3.7/site-packages/rancher.py:268: in _get
    return self._unmarshall(self._get_raw(url, data=data))
/usr/local/lib/python3.7/site-packages/rancher.py:60: in wrapped
    return fn(*args, **kw)
/usr/local/lib/python3.7/site-packages/rancher.py:275: in _get_raw
    r = self._get_response(url, data)
/usr/local/lib/python3.7/site-packages/rancher.py:282: in _get_response
    self._error(r.text)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rancher.Client object at 0x7f50e24e9748>
text = '{"baseType":"error","code":"Forbidden","message":"clusters.management.cattle.io \\"c-lf944\\" is forbidden: User \\"u...t resource \\"clusters\\" in API group \\"management.cattle.io\\" at the cluster scope","status":403,"type":"error"}\n'

    def _error(self, text):
>       raise ApiError(self._unmarshall(text))
E       rancher.ApiError: (ApiError(...), 'Forbidden : clusters.management.cattle.io "c-lf944" is forbidden: User "u-v92fp" cannot get resource "clusters" in API group "management.cattle.io" at the cluster scope\n\t{\'baseType\': \'error\', \'code\': \'Forbidden\', \'message\': \'clusters.management.cattle.io "c-lf944" is forbidden: User "u-v92fp" cannot get resource "clusters" in API group "management.cattle.io" at the cluster scope\', \'status\': 403, \'type\': \'error\'}')
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master-head with ECM enabled (5c1c2c5)
- Installation option (single install/HA): HA

**Comment by westlywright** [comment]
The UI specifically waits for the condition `InitialRolesPopulated=true` because certain clusters have actions that have to happen after this condition is met. The error is a valid. Normally after cluster creation you'd sit on this page very briefly as the `InitialRolesPopulated` flag is set and the change comes in over the web socket event bus. This flag was not set in the time frame (30 seconds) and the UI reports that. This is NOT a UI bug.

But this does bring up the question as to what we should do in this case? I believe there could be some additional usability work done here.

The Bigger issue is why that flag is failing to be set? The UI has no control over this.

**Comment by sowmyav27** [comment]
See a similar error while deploying clusters in an upgraded setup. Issue logged - https://github.com/rancher/rancher/issues/28254

**Comment by sowmyav27** [comment]
See the same error when deploying an EKS cluster (Not the KEv2 EKS, the eks on 2.4.x) through API on master-head commit id: `9b0dd20b7`
Error seen in the automation job: 
```
    def _error(self, text):
>       raise ApiError(self._unmarshall(text))
E       rancher.ApiError: (ApiError(...), 'Forbidden : clusters.management.cattle.io "c-ct8gr" is forbidden: User "u-hhbrd" cannot get resource "clusters" in API group "management.cattle.io" at the cluster scope\n\t{\'baseType\': \'error\', \'code\': \'Forbidden\', \'message\': \'clusters.management.cattle.io "c-ct8gr" is forbidden: User "u-hhbrd" cannot get resource "clusters" in API group "management.cattle.io" at the cluster scope\', \'status\': 403, \'type\': \'error\'}')
```

**Comment by aaronRancher** [comment]
The bug is reproduced on an upgrade to Rancher HA server `rancher\rancher:v2.5.0-alpha1`

Steps:
Provision a `v2.4.6` Rancher server
Provision an AKS cluster and GKE cluster
After clusters become available, upgrade setup to `v2.5.0-alpha1` 
Provision another AKS cluster and GKE cluster

Result:
Hit the same error when trying to create clusters.
![Screen Shot 2020-08-28 at 3 43 18 PM](https://user-images.githubusercontent.com/57409052/91621237-46eb5880-e947-11ea-810c-86d29c7dadbb.png)
After trying again, observed a different error:
![Screen Shot 2020-08-28 at 3 42 13 PM](https://user-images.githubusercontent.com/57409052/91621271-62eefa00-e947-11ea-801e-c11d64437e1a.png)

**Comment by soumyalj** [comment]
Tested with master-head(master-76c26fb0f9d4)

- Create a GKE, AKS cluster
- After hitting "Save" button, the UI is not stuck in "Saving" state. Provisioning starts with out any errors.
